compiler/rustc_resolve/src/late.rs

```diff
@@ -4702,6 +4714,39 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
             });
         }
     }
+
+    fn record_where_bound_predicate_with_poly_trait(&mut self, p: &WherePredicate) {
+        if let ast::WherePredicate::BoundPredicate(ast::WhereBoundPredicate {
+            bounded_ty,
+            bounds,
+            ..
+        }) = p
+        {
+            for bound in bounds {
+                if let ast::GenericBound::Trait(
+                    ast::PolyTraitRef {
+                        span: poly_span,
+                        trait_ref: ast::TraitRef { path: ast::Path { span, .. }, .. },
```

Comment:
style nit for the future: when you have a struct, prefer using `trait_ref.path.span` instead of destructuring one field

Reply:
> style nit for the future: when you have a struct, prefer using `trait_ref.path.span` instead of destructuring one field

Done. Thank you.